### ISSUE TITLE
Polyfills for dependencies

### ISSUE THREAD

**Issue by jamiebuilds** [sindresorhus/ponyfill#4]
@ljharb brought this up in TC39 the other day...

Ponyfills should absolutely be encouraged, I think there's plenty of benefits to using them. However, there's a valid use case of polyfills that gets ignored:

When you have nested dependencies that have less backwards support than you do they won't use ponyfills/polyfills. Now you can't use that particular dependency without using a full polyfill unless you fork the dependency (which could also be several deps deep).
1. I think package authors should be encouraged to use ponyfills longer than everyone else
2. I think polyfills should be limitedly encouraged as the correct solution to that particular problem

**Comment by sindresorhus** [comment]
Agreed

**Comment by lacymorrow** [comment]
Yes, I will say that Ponyfills are quite clever for graceful degradation. Support everything. It's our job.
